lib/src/feature/articles/view/article_search_screen.dart

```diff
@@ -0,0 +1,249 @@
+            style: AppTextStyle.titleTextStyle
+                .copyWith(fontSize: AppFontSize.f16)),
+      ],
+    );
+  }
+
+  Widget _renderSearchFieldAndTag() {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: AppPadding.p16),
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          XTextFieldWithLabel(
+            hintText: S.of(context).enterArticlesName,
+            hintStyle: AppTextStyle.hintTextStyle,
+            suffix: const Icon(
+              Icons.search,
+              color: AppColors.primary,
+            ),
+            onChanged: (text) {
+              _onChangeText(text);
+            },
+          ),
+          BlocBuilder<ArticlesSearchBloc, ArticlesSearchState>(
+              buildWhen: (pre, cur) =>
+                  listEquals(pre.listSelectedTags, cur.listSelectedTags) ==
+                      false ||
+                  pre.isExpand != cur.isExpand ||
+                  pre.status != cur.status ||
+                  listEquals(pre.tags, cur.tags) == false,
+              builder: (context, state) {
+                return Wrap(
+                  spacing: AppPadding.p6,
+                  runSpacing: AppPadding.p6,
+                  children: [
+                    for (String tag in state.tags ?? [])
+                      XWarningChip(
+                        title: tag,
+                        hasIcon: false,
+                        border: state.listSelectedTags?.contains(tag) == true
+                            ? Border.all(width: 1.5, color: AppColors.primary)
+                            : Border.all(width: 0.5, color: AppColors.black),
+                        bgColor: state.listSelectedTags?.contains(tag) == true
+                            ? AppColors.subPrimary
+                            : AppColors.white,
+                        onTap: (tag) => context
+                            .read<ArticlesSearchBloc>()
+                            .setListSelectedTag(tag),
+                      )
+                  ],
+                );
+              })
+        ],
+      ),
+    );
+  }
+
+  Widget _renderArticles() {
+    return Expanded(
+        child: BlocSelector<ArticlesSearchBloc, ArticlesSearchState,
+            List<MArticles>?>(
+      selector: (state) => state.listSearchedArticles,
+      builder: (context, articles) {
+        return isNullOrEmpty(articles) ||
+                isNullOrEmpty(
+                    context.read<ArticlesSearchBloc>().state.listSearchedImage)
+            ? const SizedBox.shrink()
+            : ListView.builder(
+                itemCount: articles!.length,
+                itemBuilder: (context, index) => _renderArticleCard(
+                    context,
+                    articles[index],
+                    context
+                        .read<ArticlesSearchBloc>()
+                        .state
+                        .listSearchedImage![articles[index].id]));
```

Comment:
check null safety, don't using "!". It so risk

Reply:
has checked is not null above